plone/app/event/dx/behaviors.py

```diff
@@ -339,7 +340,23 @@ def uid(self):
 
     @property
     def url(self):
-        return safe_text(self.context.absolute_url())
+        """calculate the path, is required as long the ram.cache is active in portlets renderer
+        the return value of self.context.absolute_url() differs
+        with ram cache: portal/testtermin
+        without ram cache: http://site.local/testevent
+        """
+
+        portal = getSite()
+        portal_url = portal.absolute_url()
+        portal_path = list(portal.getPhysicalPath())
+        event_path = list(self.context.getPhysicalPath())
+        path_without_portal = ""
+        if len(portal_path) > 0:
+            path_without_portal = "/".join(event_path[len(portal_path) :])
```

Comment:
Question: in your issue you've mentioned the comments in `OFS.Traversal` ... there the method `absolute_url_path()` is mentioned. Would this help here too? Something like

```
f"{portal.absolute_url()}/{self.context.absolute_url_path()}"
```

https://github.com/zopefoundation/Zope/blob/c33b799be066e3e72d6d3ef0b37e0b63199ed9bf/src/OFS/Traversable.py#L79

Reply:
No, the portal id is in the path segments and should be removed. Background: i have configured WEBDAV via apache2
```
# apache2 WEBDAV
<Location /my-site/>
  Dav On
  ProxyPass http://127.0.0.1:9090/my-site/
  ProxyPassReverse http://127.0.0.1:9090/my-site/
</Location>
```
``` # buildout.cfg
[client1]
<= client_base
http-address    = ${buildout:ip_base}:${buildout:client_port_base}81 0.0.0.0:9090
zope-conf-additional +=
  webdav-source-port 9090
  http-realm Welcome at Zeo Client1
```

The URL which is generate in the portlet, point to the WEBDAV Location on the second load. The portal id is not needed in the url.

Reply:
Sorry for asking. But it is strange that this change is needed at all. `self.context.absolute_url()` should always be an URL not a path as you mentioned in your issue. It might be a deeper problem in the caching logic introduced here https://github.com/plone/plone.app.event/commit/15696d9b0f35e255520c618ae22bb2c4a51b0919 from @frapell which absolutely makes sense in terms of performance.

Reply:
No need to apologise.

first call without caching:

`portal.absolute_url():  http://localhost:8080/my-site`
`self.context.absolute_url():  http://localhost:8080/my-site/my-event`
`self.context.absolute_url(True):  my-site/my-event`
`self.context.absolute_url_path():  /my-site/my-event`

second call with cache

`portal.absolute_url():  http://localhost:8080/my-site`
`self.context.absolute_url():  /my-site/my-event`
`self.context.absolute_url(True):  my-site/my-event`
`self.context.absolute_url_path():  /my-site/my-event`

Reply:
the changes from @frapell calculates only the cache key. for me it feels right, that no hostname is in the url, if the object come from ram cache. but it's only a feeling not evidence. I'm not a Zope Guru ;-)